### PR TITLE
Add provenance field docs for the GitHub hosted build

### PR DESCRIPTION
**Pull request by asraa**
Fixes https://github.com/slsa-framework/slsa-github-generator/issues/166

### CODE REVIEW

**Comment by asraa** [comment]
Formatting this into a table was really difficult because of the embedded examples :/

Adding the release tag though

File: README.md [hunk 1]
```diff
@@ -31,3 +31,84 @@ Find our blog post series [here](https://security.googleblog.com/2022/04/improvi
...
+    "entryPoint": "Go SLSA Release"
+},
+```
+`Invocation.parameters`: This describes any parameters from trigger events.
+
+`Invocation.environment`: This describes the GitHub workflow environment, including the event information.
+```json
+"environment": {
+    "arch": "X64",
+    "github_actor": "laurentsimon",
```
Comment (laurentsimon) [inline README.md:69]: shall we add some comments about each field here?
I also added support repository_id, repository_owner_id and actor_id.

File: README.md [hunk 2]
```diff
@@ -31,3 +31,84 @@ Find our blog post series [here](https://security.googleblog.com/2022/04/improvi
...
+    "github_event_name": "workflow_dispatch",
+    "github_event_payload": {
+      "inputs": null,
+      "ref": "refs/heads/main",
+      "repository": { ... }
+    },
+}
+```
+
+`BuildConfig`: This contains information on the steps of the build. The default is nil, specific builders implement their own.
```
Comment (laurentsimon) [inline README.md:80]: shall we link to a build-specific file with a similar doc, but focusing on the builder's BuildConfig fields?

**Review by laurentsimon — APPROVED** [review]
Thanks!

File: README.md (hunk 2 above)
Comment (asraa) [reply README.md:80]: Nice, I'll add docs on the `BuildConfig` for the go builder and link here

File: README.md (hunk 1 above)
Comment (asraa) [reply README.md:88]: done

**Comment by ianlewis** [comment]
> Formatting this into a table was really difficult because of the embedded examples :/

If you're talking about formatting the markdown text, formatting tools really help. I've found that https://prettier.io works pretty well.

**Pull request merged**